build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -637,60 +622,57 @@ export const handlers: Handler[] = [
 				: undefined;
 		},
 		resolver: (file) => {
-			let json;
-			try {
-				json = JSON.parse(readFile(file));
-			} catch (err) {
-				return { resolved: false, message: "Error parsing JSON file: " + file };
-			}
+			updatePackageJsonFile(path.dirname(file), (json) => {
+				const hasScriptsField = Object.prototype.hasOwnProperty.call(json, "scripts");
 
-			const resolved = true;
+				if (hasScriptsField) {
+					const hasFormatScriptResolver = Object.prototype.hasOwnProperty.call(
+						json.scripts,
+						"format",
+					);
 
-			const hasScriptsField = Object.prototype.hasOwnProperty.call(json, "scripts");
+					const hasPrettierScriptResolver = Object.prototype.hasOwnProperty.call(
+						json.scripts,
+						"prettier",
+					);
 
-			if (hasScriptsField) {
-				addPrettier(json);
-				writeFile(file, JSON.stringify(sortPackageJson(json), undefined, 2) + newline);
-			}
+					const hasPrettierFixScriptResolver = Object.prototype.hasOwnProperty.call(
+						json.scripts,
+						"prettier:fix",
+					);
 
-			return { resolved: resolved };
+					if (
+						hasFormatScriptResolver ||
+						hasPrettierScriptResolver ||
+						hasPrettierFixScriptResolver
+					) {
+						const formatScript = json["scripts"]["format"]?.includes("lerna");
```

Comment:
you can just do

```suggestion
						const formatScript = json.scripts.format?.includes("lerna");
```
Same cleanup below as well.